src/lib/pubkey/classic_mceliece/cmce_poly.h

```diff
@@ -0,0 +1,174 @@
+       */
+      const Classic_McEliece_GF& coef_at(size_t i) const { return m_coef.at(i); }
+
+      /**
+       * @brief Get the entire coefficients vector of the polynomial.
+       */
+      const std::vector<Classic_McEliece_GF>& coef() const { return m_coef; }
+
+      /**
+       * @brief Get the degree of the polynomial.
+       *
+       * Note that the degree is given by the size of the coefficient vector, even if the leading coefficient is zero.
+       */
+      size_t degree() const { return m_coef.size() + 1; }
+
+   private:
+      std::vector<Classic_McEliece_GF> m_coef;
+};
+
+/**
+ * @brief Representation of a minimal polynomial in GF(q)[y].
+ *
+ * It represents the monic irreducible degree-t polynomial of the goppa code.
+ */
+class BOTAN_TEST_API Classic_McEliece_Minimal_Polynomial : public Classic_McEliece_Polynomial {
+   public:
+      Classic_McEliece_Minimal_Polynomial(std::vector<Classic_McEliece_GF> coef) :
+            Classic_McEliece_Polynomial(std::move(coef)) {}
+
+      /**
+       * @brief Serialize the polynomial to bytes according to ISO Section 9.2.9.
+       */
+      secure_vector<uint8_t> serialize() const;
+
+      /**
+       * @brief Create a polynomial from bytes according to ISO Section 9.2.9.
+       */
+      static Classic_McEliece_Minimal_Polynomial from_bytes(std::span<const uint8_t> bytes, GF_Mod poly_f);
+};
+
+// Stores all auxiliary information and logic of FF_(q^t) via FF_q[y]/F(y)
+/**
+ * @brief Represents the polynomial ring GF(q)[y]/F(y) where F(y) is the modulus polynomial in
+ * GF(q)[y] of degree t.
+ *
+ * This class contains a modulus polynomial F(y) and the GF(q) modulus f(z). It is used
+ * to create and operate with Classic_McEliece_Polynomials.
+ *
+ */
+class BOTAN_TEST_API Classic_McEliece_Polynomial_Ring {
+   public:
+      /**
+       * @brief Represents a non-zero coefficient of the modulus F(y) (which is in GF(q)[y]).
+       *
+       * E.g. {.idx = 4, .coeff = (z+1)} represents the monomial (z+1)y^4.
+       */
+      struct BOTAN_TEST_API Big_F_Coefficient {
+            size_t idx;
+            Classic_McEliece_GF coeff;
+      };
+
+      /**
+       * @brief Construct a polynomial ring GF(q)[y]/F(y) by defining the polynomial modulus F(y),
+       * the GF(q) modulus f(z) and the degree of F(y).
+       *
+       * F(y) is given by a vector of Big_F_Coefficients, where each one represents a monomial of F(y).
+       * However, the highest monomial must not be specified, since it is always 1.
+       *
+       * @param poly_big_f_coef The non-zero coefficients of F(y) in GF(q)[y] WITHOUT the highest monomial.
+       * @param poly_f The modulus f(z) of GF(q).
+       * @param t The polynomial degree of the ring (and of F(y)).
+       */
+      Classic_McEliece_Polynomial_Ring(const std::vector<Big_F_Coefficient>& poly_big_f_coef, GF_Mod poly_f, size_t t) :
+            m_position_map(poly_big_f_coef), m_t(t), m_poly_f(poly_f) {}
```

Comment:
```suggestion
      Classic_McEliece_Polynomial_Ring(std::vector<Big_F_Coefficient> poly_big_f_coef, GF_Mod poly_f, size_t t) :
            m_position_map(std::move(poly_big_f_coef)), m_t(t), m_poly_f(poly_f) {}
```

... the only place I found, this is called with an rvalue of the vector and will benefit form this.